garage/misc/wrappers.py

```diff
@@ -0,0 +1,88 @@
+"""Gym env wrappers"""
+import cv2
+import gym
+import numpy as np
+
+from collections import deque
+from gym.spaces import Box
+
+
+class RepeatActionWrapper(gym.core.Wrapper):
+    """Gym env wrapper for repeating action for n frames"""
+
+    def __init__(self, env, frame_to_repeat):
+        super(RepeatActionWrapper, self).__init__(env)
+        self.frame_to_repeat = frame_to_repeat
+
+    def step(self, action):
+        for i in range(self.frame_to_repeat):
+            obs, reward, done, info = self.env.step(action)
+        return obs, reward, done, info
+
+    def reset(self):
+        return self.env.reset()
+
+
+class ResizeAndGrayscaleWrapper(gym.core.Wrapper):
+    """Wrapper for resizing to (w, h) and converting frames to grayscale."""
+
+    def __init__(self, env, w, h, plot=False):
+        super(ResizeAndGrayscaleWrapper, self).__init__(env)
+        self.observation_space = Box(0.0, 1.0, shape=[w, h], dtype=np.float32)
+        self.w = w
+        self.h = h
+        self.plot = plot
+
+    def _observation(self, obs):
+        obs = cv2.cvtColor(obs, cv2.COLOR_RGB2GRAY)
+        obs = cv2.resize(obs, (self.w, self.h), interpolation=cv2.INTER_AREA)
+        obs = obs.astype(np.float32) / 255.0
+        if self.plot:
```

Comment:
You should not be plotting inside some random helper function. The only valid place to plot is when someone calls `render()`. This also means your `plot` argument is irrelevant.